railties/lib/rails/generators.rb

```diff
@@ -156,12 +156,25 @@ def self.invoke(namespace, args=ARGV, config={})
         args << "--help" if args.empty? && klass.arguments.any?(&:required?)
         klass.start(args, config)
       else
-        options     = sorted_groups.map(&:last).flatten
-        suggestions = options.sort_by {|suggested| levenshtein_distance(namespace.to_s, suggested) }.first(3)
-        msg =  "Could not find generator '#{namespace}'. "
-        msg << "Maybe you meant #{ suggestions.map {|s| "'#{s}'"}.join(" or ") }\n"
-        msg << "Run `rails generate --help` for more options."
-        puts msg
+        unless sorted_groups.group_by(&:size).min.first > namespace.length
```

Comment:
This is a confusing line of code, we're saying skip the suggestion if the longest option is less than the provided option (i.e. `if longest_option < namespace.length `) ? We generally prefer if's with the exception of inline logic.

I don't think it's needed, and fails in the case someone is actually looking for the longest option, and accidentally enters an extra character.